### ISSUE TITLE
New Rafactoring: Create function

### ISSUE THREAD

**Issue by dutzi** [nicoespeon/abracadabra#64]
A lot of times I would write:

```jsx
return <button onClick={handleClick}/>
```

And then, above the return statement I would implement handleClick.

I would be nice if when the cursor is within the identifier `handleClick`, I could have the option to create a function with the same name, resulting in the following code:

```jsx
function handleClick() {

}

return <button onClick={handleClick}/>
```

And just wanted to add that I like this extension a lot! "Extract Variable" especially has been very useful in my day to day coding.

**Comment by nicoespeon** [comment]
Hi @dutzi 👋 
First of all, I wish you a happy new year!

Thanks for the suggestion. Indeed, I often found myself in that kind of situation so I can relate.
Hence, I've a good news, but also a bad news 😅 

The good news is: **I already created an extension to handle precisely that**. Have a look at [Hocus Pocus](https://github.com/nicoespeon/hocus-pocus).

The bad news is: **I just checked and this exact operation won't work**. It will only try to create a variable, and it will not create it properly because of the JSX context. That's worth creating issues in the Hocus Pocus repo, so I just did that: https://github.com/nicoespeon/hocus-pocus/issues

Thus, I'm closing this issue in favor of the 2 I created on Hocus Pocus 😉 

_Why another extension and not just another refactoring in Abracadabra directly? Because of the scope of Abracadabra. I don't want to open the door to other operations than refactorings. Creating an undeclared function is not a refactoring because it changes the way the code works._
